sec1_intro_and_setup/test_app/db/seeds.rb

```diff
@@ -0,0 +1,7 @@
+# This file should contain all the record creation needed to seed the database with its default values.
```

Comment:
Whats this file for?

Reply:
`Seeds` let you fill a database with data. This file is where the seed data is stored.

Reply:
What is `seed data`?

Reply:
Do you mean once I have the database and data, i cannot use seed file to add more data in?

Reply:
No, you still can add more data but not into the seed file but the seed folder, for example `db/seeds/ ...` where `...` can be several `<file_name>.rb` files.  Since rails support initial data creation inside the `db/seed.rb` file, it will get larger and complex as the application developed, so we can group the `<file_name>.rb` files into the `db/seeds/...` folder and load them into the `seeds.rb` file.

Reply:
Why do you think you can't?
Whats stopping you from adding more data into seeds.rb file?

Reply:
Let say we add the data all into `seed.rb`, then the file will be very large and it takes a longer time to load the data to the database and to update the seed file.

Reply:
But you can do it isn't it?
Its just not good to do it. There is nothing stopping you from doing it.

Thats one. Secondly, so what if the file is super big and takes a while to load all required data into the database? So what it is time consuming? Seed is generally for testing environment (staging) or your own local. I would personally trade time than having zero data in a big system. 

Lets say the system is big. Big to the point where by you are in a team that handles one small part of the system. However, the feature you need to debug is done by another team. Instead of immediately focusing your time on fixing the problem, you now have to figure out how to get the data in your local environment to debug the issue that is reported in production. Figuring out how to get the data takes a lot of time & effort especially if you aren't the one who build it or if you have no idea about that part of the system.

Reply:
> But you can do it isn't it?

You can do it, just that it takes time to load the seed to a specific model based on different needs if you put all seeds into one file. 
(In case you ask what is a specific model based on different needs: 
Let say we are having an e-commerce application and we develop a new feature, we need both users and sellers to feedback to us on the feature in the testing stage, we have 2 models here: a user model, and a seller model, both need different seed data)

> what it is time-consuming?

Loading all data every time although you just need the data for one model (instead, you load all seed data whether the seed data is related to the model you specified since you put all seed data into one file)

>  what if the file is super big and takes a while to load all required data into the database?

You can separate the seed based on different needs, so the process can be faster and the seed is easier to manage (updating seed data, managing duplicated data etc).

Reply:
![image](https://user-images.githubusercontent.com/5483026/129812152-fa26a03f-8230-43c5-ad79-e3d9a2799ef0.png) So, can you or can't you? Be sure of your stance please.

> Loading all data every time although you just need the data for one model (instead, you load all seed data whether the seed data is related to the model you specified since you put all seed data into one file)
What do you mean by loading all data every time?